save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/basic/contests/ContestCreationComponent.kt

```diff
@@ -0,0 +1,214 @@
+import com.saveourtool.save.frontend.externals.modal.Styles
+import com.saveourtool.save.frontend.externals.modal.modal
+import com.saveourtool.save.frontend.utils.*
+import com.saveourtool.save.frontend.utils.noopLoadingHandler
+import com.saveourtool.save.utils.LocalDateTime
+
+import csstype.ClassName
+import org.w3c.fetch.Headers
+import org.w3c.fetch.Response
+import react.*
+import react.dom.html.ButtonType
+import react.dom.html.ReactHTML.button
+import react.dom.html.ReactHTML.div
+import react.dom.html.ReactHTML.form
+
+import kotlin.js.json
+import kotlinx.browser.window
+import kotlinx.datetime.LocalDate
+import kotlinx.datetime.LocalTime
+import kotlinx.serialization.encodeToString
+import kotlinx.serialization.json.Json
+
+/**
+ * Component that allows to create new contests
+ */
+val contestCreationComponent = contestCreationComponent()
+
+private val contestCreationCard = cardComponent()
+
+/**
+ *  Contest creation component props
+ */
+external interface ContestCreationComponentProps : Props {
+    /**
+     * Name of current organization
+     */
+    var organizationName: String
+
+    /**
+     * Callback invoked on successful contest creation
+     */
+    var onSaveSuccess: (String) -> Unit
+
+    /**
+     * Callback invoked on error while contest creation
+     */
+    var onSaveError: (Response) -> Unit
+}
+
+/**
+ * @param organizationName name of an organization to which a new contest will be linked
+ * @param isOpen flag that indicates if the modal is open
+ * @param onSuccess callback invoked on successful contest creation
+ * @param onFailure callback invoked on error when creating contest
+ * @param onClose callback invoked on close button press
+ */
+fun ChildrenBuilder.showContestCreationModal(
+    organizationName: String,
+    isOpen: Boolean,
+    onSuccess: (String) -> Unit,
+    onFailure: (Response) -> Unit,
+    onClose: () -> Unit,
+) {
+    modal { props ->
+        props.isOpen = isOpen
+        props.style = Styles(
+            content = json(
+                "top" to "25%",
+                "left" to "30%",
+                "right" to "30%",
+                "bottom" to "auto",
+                "position" to "absolute",
+                "overflow" to "hide"
+            ).unsafeCast<CssProperties>()
+        )
+        contestCreationComponent {
+            this.organizationName = organizationName
+            onSaveSuccess = onSuccess
+            onSaveError = onFailure
+        }
+        div {
+            className = ClassName("d-flex justify-content-center")
+            button {
+                type = ButtonType.button
+                className = ClassName("btn btn-secondary mt-4")
+                +"Cancel"
+                onClick = {
+                    onClose()
+                }
+            }
+        }
+    }
+}
+
+private fun String.toLocalDateTime(time: LocalTime = LocalTime(0, 0, 0)) = LocalDateTime(LocalDate.parse(this), time)
```

Comment:
Maybe `dateToLocalDateTime`? Seems confusing that `this` contains only date, while method name implies conversion to datetime

Reply:
Makes sense, will do it